### PR TITLE
Verify password for local-account activation

### PR DESCRIPTION
**Pull request by 6543**
This is to prevent 3rd party activation

source: https://codeberg.org/Codeberg/gitea/commit/cbbf29d76d55067e3a3a55a17a6a96a78c90f35a

### CODE REVIEW

**Comment by gary-kim** [comment]
@6543 Mind taking a look at the linter error?

File: models/user.go [hunk 1]
```diff
@@ -890,6 +890,16 @@ func VerifyUserActiveCode(code string) (user *User) {
 	return nil
 }
 
+// VerifyUserActiveCode verifies active code and password when activating account
```
Comment (lafriks) [inline models/user.go]:
```suggestion
// VerifyUserActiveCodeAndPassword verifies active code and password when activating account
```

**Comment by 6543** [comment]
@gary-kim lafriks fixed it for us :) - ready for a look again

**Comment by zeripath** [comment]
Does account activation possibly occur with OAUTH2 based users?

**Comment by 6543** [comment]
@ashimokawa ping ;)

File: templates/user/auth/activate.tmpl [hunk 2]
```diff
@@ -18,7 +18,20 @@
...
 						{{end}}
 					{{else}}
-						{{if .IsSendRegisterMail}}
+						{{if .NeedsPassword}}
+							<form class="ui form" action="/user/activate" method="post">
+								<div class="required inline field">
+									<label for="password">{{.i18n.Tr "password"}}</label>
+									<input id="password" name="password" type="password" autocomplete="off" required>
+								</div>
+
```
Comment (silverwind) [inline templates/user/auth/activate.tmpl]:
```suggestion
```

**Review by zeripath — APPROVED** [review]

**Review by silverwind — APPROVED** [review]

**Comment by techknowlogick** [comment]
🚀

**Pull request merged**